### PR TITLE
LUN-398: Edit mode for own vocabulary

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

**Review by steffenkleinle — APPROVED** [review]
It is not really possible to test this yet, is it?

**Review by f1sh1918 — APPROVED** [review]
Tested on ios. looks nice. great work :)

**Comment by SteffiStuffel** [comment]
> It is not really possible to test this yet, is it?

Yes you can test, after adding dummy data by adding `AsyncStorage.addUserDocument(new DocumentBuilder(4).build()[0])`

File: src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx [hunk 1]
```diff
@@ -41,17 +44,50 @@ interface Props {
...
+  const openDeleteConfirmationModal = (document: Document): void => {
+    setModalVisible(true)
+    setDocumentToDelete(document)
+  }
+
+  const deleteItem = (document: Document | null): void => {
+    if (document) {
+      AsyncStorage.deleteUserDocument(document)
+        .then(documents.refresh)
+        .catch(err => reportError(err))
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/routes/user-vocabulary-list/UserVocabularyListScreen.tsx:74]: aaarg, why is it impossible for my brain to do this :D

File: src/services/AsyncStorage.ts [hunk 2]
```diff
@@ -153,8 +153,10 @@ const setUserVocabulary = async (userDocument: Document[]): Promise<void> => {
 }
 
 const addUserDocument = async (userDocument: Document): Promise<void> => {
-  console.log('add: ', userDocument.word)
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/services/AsyncStorage.ts:156]: console.log is only a warning, i guess we should change this. Will create a ticket.

File: src/constants/labels.json [hunk 3]
```diff
@@ -189,7 +189,11 @@
     "list": {
       "edit": "Bearbeiten",
       "finished": "Fertig",
-      "noWordsYet": "Du hast noch keine eigenen Wörter erstellt."
+      "noWordsYet": "Du hast noch keine eigenen Wörter erstellt.",
+      "confirmDeletionPart1": "Willst du das Wort",
+      "confirmDeletionPart2": "wirklich löschen?",
+      "confirm": "Ja, löschen",
+      "cancel": "Schliessen"
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/constants/labels.json:196]: As this is changed to uppercase, i think "ss" makes more sense than "ß", also UI/UX suggested "ss". Will keep.

File: src/services/AsyncStorage.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply src/services/AsyncStorage.ts:156]: 👍

**Pull request merged**